token/program-2022/src/offchain.rs

```diff
@@ -33,10 +36,18 @@ use {
 ///     &mint,
 /// ).await?;
 /// ```
+/// Note that this offchain helper will build a new `Execute` instruction,
+/// resolve the extra account metas, and then add them to the transfer
+/// instruction. This is because the extra account metas are configured
+/// specifically for the `Execute` instruction, which requires five accounts
+/// (source, mint, destination, authority, and validation state), wheras the
+/// transfer instruction only requires four (source, mint, destination, and
+/// authority) in addition to `n` number of multisig authorities.
 pub async fn resolve_extra_transfer_account_metas<F, Fut>(
     instruction: &mut Instruction,
     fetch_account_data_fn: F,
     mint_address: &Pubkey,
+    amount: u64,
 ) -> Result<(), AccountFetchError>
```

Comment:
Since this is specific to transfers, let's deprecate this function, and create a new one which requires passing in all the transfer parameters, and that way we don't have to assume that a proper transfer instruction is provided.